framework/base/Model.php

```diff
@@ -601,26 +636,32 @@ public function getFirstError($attribute)
     }
 
     /**
-     * Adds a new error to the specified attribute.
-     * @param string $attribute attribute name
+     * Adds a new error to the specified attribute or common error.
+     * @param string $attribute attribute name. Leave `null` for common errors (not related with particular attribute).
      * @param string $error new error message
+     * @throws InvalidParamException in case of attribute not exist
      */
-    public function addError($attribute, $error = '')
+    public function addError($attribute = null, $error = '')
```

Comment:
Every change of public or protected method signature is BC break since you have to change all the classes overriding it when it changes.